source/faucet/main.d

```diff
@@ -629,15 +695,21 @@ private const(string)[] parseSequence (string section,
 }
 
 /// Parse the config file
-public const(string)[] parseConfigFile (string config_path)
+public Config parseConfigFile (string configPath)
 {
-    Node root = Loader.fromFile(config_path).load();
-    return parseSequence("keys", root);
+    Node root = Loader.fromFile(configPath).load();
+    return Config(
+        Generator(root["generator"]),
+        root["web_bind_address"].as!string,
+        root["web_bind_port"].as!ushort,
+        root["stats_port"].as!ushort);
 }
 
 /// Global because we need to access it from our signal handler
 private Faucet inst;
 
+private SecretKey[PublicKey] secret_keys;
```

Comment:
This should be in the upper part of the file because the variable is being used in the above part of the file.

Reply:
Oh why did this become global?

Reply:
I update them before creating the Faucet instance.